### ISSUE TITLE
Can not mix '--config' with arguments [pod-network-cidr service-cidr]

### ISSUE THREAD

**Issue by gunsluo** [kubernetes/kubeadm#909]
## Is this a BUG REPORT or FEATURE REQUEST?
BUG REPORT

## Versions

**kubeadm version** (use `kubeadm version`):
kubeadm version: &version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:00:59Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

**Environment**:
- **Kubernetes version** (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.4", GitCommit:"5ca598b4ba5abb89bb773071ce452e33fb66339d", GitTreeState:"clean", BuildDate:"2018-06-06T08:13:03Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"linux/amd64"}

- **Cloud provider or hardware configuration**:
- **OS** (e.g. from /etc/os-release):
```
NAME="CentOS Linux"
VERSION="7 (Core)"
ID="centos"
ID_LIKE="rhel fedora"
VERSION_ID="7"
PRETTY_NAME="CentOS Linux 7 (Core)"
ANSI_COLOR="0;31"
CPE_NAME="cpe:/o:centos:centos:7"
HOME_URL="https://www.centos.org/"
BUG_REPORT_URL="https://bugs.centos.org/"

CENTOS_MANTISBT_PROJECT="CentOS-7"
CENTOS_MANTISBT_PROJECT_VERSION="7"
REDHAT_SUPPORT_PRODUCT="centos"
REDHAT_SUPPORT_PRODUCT_VERSION="7"
```

- **Kernel** (e.g. `uname -a`):
Linux localhost.localdomain 3.10.0-862.3.2.el7.x86_64 #1 SMP Mon May 21 23:36:36 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

## What happened?
run `kubeadm init --kubernetes-version v1.10.4 --service-cidr 10.96.0.0/16 --pod-network-cidr 10.244.0.0/16 --apiserver-advertise-address 10.0.100.2`

`etcd` cannot start normally.   err: [etcdmain: listen tcp 183.136.168.82:2380: bind: cannot assign requested address], 183.136.168.82 is not localhost ip, I don't know where it was. 

related to the issue:
- https://github.com/coreos/etcd/issues/9070
- https://github.com/kubernetes/kubernetes/issues/57709

 so i need to set `listen-peer-urls` .
new command : `kubeadm init --config kubeadm.yaml --service-cidr 10.96.0.0/16 --pod-network-cidr 10.244.0.0/16`

kubeadm.yaml like:
```
apiVersion: kubeadm.k8s.io/v1alpha1
kind: MasterConfiguration
kubernetesVersion: v1.10.4
api:
  advertiseAddress: '10.0.100.2'
etcd:
  extraArgs:
    'listen-peer-urls': 'http://127.0.0.1:2380'
```

error msg : can not mix '--config' with arguments [pod-network-cidr service-cidr] 

note: `kubeadm init --config kubeadm.yaml` it works well(include etcd). 

## What you expected to happen?
I expected the etcd pod to correctly listen on a localhost IP address,  also supports the simultaneous use of `--config`/`pod-network-cidr`/`service-cidr` parameters

**Comment by stealthybox** [comment]
This seems odd.
@gunsluo can you show us the contents of `/etc/kubernetes/manifests/etcd.yaml`?

**Comment by luxas** [comment]
> kubeadm init --config kubeadm.yaml --service-cidr 10.96.0.0/16 --pod-network-cidr 10.244.0.0/16

This isn't expected to work atm. Put the service cidr and pod network cidr args in the config file instead please and see if it works

**Comment by gunsluo** [comment]
@stealthybox  @luxas 
Sorry for the late reply.  this is a network problem. It won't have any problems when I restart the network on the local virtual machine.  but I couldn't find the problem.

**Comment by Arnavion** [comment]
https://github.com/kubernetes/kubernetes/issues/57709 was duped against this, and the problem of that issue still exists. kubeadm can't provision a working cluster by default when there's a DNS entry for `localhost.<dns search path>`. The user has to set `listen-peer-urls`

Should one of these issues be reopened?

**Comment by selmison** [comment]
@Arnavion 
This issue is being addressed by PR kubernetes/kubernetes#66968

**Comment by Arnavion** [comment]
@selmison Your PR doesn't help with https://github.com/kubernetes/kubernetes/issues/57709 which is what my comment was about.
